src/faction.cpp

```diff
@@ -41,13 +41,13 @@ faction::faction()
     mapy = 0;
     size = 0;
     power = 0;
-    id = "";
-    desc = "";
+    id.clear();
+    desc.clear();
```

Comment:
And here. `id` and `desc` are created empty anyway, there is no need to manually clear them.